README.md

```diff
@@ -1,27 +1,124 @@
-* [**Make iCub walk with the MPC-CP approach using the IK framework and Position Control**](https://github.com/loc2/element_capture-point-walking/issues/3)
-* [**Make iCub walk with the reactive control approach using the IK framework and Position Control**](https://github.com/loc2/element_capture-point-walking/issues/4)
+# capture-point-walking
+The **walking-controllers** project is a _suite_ of modules for bipedal locomotion for the humanoid robot platform iCub.
+
+The suite includes:
+
+* **Walking_module**: this is the main module and it implements all the controller architecture that allows iCub to walk.
+* **Joypad_module**: this module allows using the Joypad as reference input for the trajectory generated by the Walking Module
+* **WalkingLogger_module**: an module that can be useful to dump data coming from the Walking Module
+
+# Overview
+ - [:orange_book: Some theory behind the code](#orange_book-some-theory-behind-the-code)
+ - [:page_facing_up: Dependencies](#page_facing_up-dependencies)
+ - [:hammer: Build the suite](#hammer-build-the-suite)
+ - [:computer: How to run the simulation](#computer-how-to-run-the-simulation)
+ - [:running: How to test on iCub](#running-how-to-test-on-icub)
+
+# :orange_book: Some theory behind the code
+This module allows iCub walking using the position control interface.
+It implements the following architecture
+![controller_architecture](https://user-images.githubusercontent.com/16744101/37352869-757896b4-26de-11e8-97bc-10700add7759.jpg)
+where two controller loops can be distinguished:
+* an inner ZMP-CoM control loop http://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=4359259&tag=1;
+* an outer DCM control loop:
+   * model predictive controller;
+   * reactive controller.
+
+Two different inverse kinematics solver are implemented:
+* a standard non-linear IK solver;
+* a standard QP Jacobian based IK solver.
+
+# :page_facing_up: Dependencies
+* [YARP](http://www.yarp.it/): to handle the comunication with the robot;
+* [osqp-eigen](https://github.com/robotology/osqp-eigen): to solve the MPC problem;
+* [qpOASES](https://github.com/robotology-dependencies/qpOASES): to solve the IK problem;
+* [Unicycle footstep planner](https://github.com/robotology/unicycle-footstep-planner/tree/dcmTrajectoryGenerator): to generate a trajectory for the DCM;
+* [Gazebo](http://gazebosim.org/): for the simulation (tested Gazebo 8).
+
+# :hammer: Build the suite
+## Linux/macOs
+
+```sh
+git clone https://github.com/robotology/walking-controllers.git
+cd walking-controllers
+mkdir build && cd build
+cmake ../
+make
+[sudo] make install
+```
+Notice: `sudo` is not necessary if you specify the `CMAKE_INSTALL_PREFIX`. In this case it is necessary to add in the `.bashrc` or `.bash_profile` the following lines:
+```sh
+export walkingControllers_DIR=/path/where/you/installed/
```

Comment:
Maybe it is better to match the name of the project with the the prefix of this variable. Thus, I think there are two possibilities:
- change this to ``walking-controllers_DIR``
- change the project name to camel case, i.e. ``WalkingControllers`` and change this to ``WalkingControllers_DIR``

Reply:
Note that at the moment that variable is unused, because this repo does not install any CMake configuration file, but I agree that it make sense to already have it with the right name.

Reply:
Ok, Is there a coding convention regarding this?

Reply:
For simplicity I prefer to keep the name consistent between the repo and the package (i.e. `walking-controllers_DIR`), but from the past we inherited different choices (`YARP` and `yarp`, `iDynTree` and `idyntree`) some of which are quite confusing (`ICUB` and `icub-main`) so I can't say we have a convention in place for this.

Reply:
ok, the name of the environmental variable will be changed into `walking-controllers_DIR`.